app/src/sharedTest/java/org/oppia/android/app/options/AudioLanguageFragmentTest.kt

```diff
@@ -290,12 +290,70 @@ class AudioLanguageFragmentTest {
+    launchV2FlowWithLanguage(ENGLISH_AUDIO_LANGUAGE, OPTIONS_SCREEN).use {
+      onView(withText("In Oppia, you can listen to lessons!")).check(matches(isDisplayed()))
+      onView(withText(context.getString(R.string.audio_language_fragment_subtitle)))
+        .check(matches(isDisplayed()))
+      onView(withId(R.id.audio_language_dropdown_list)).check(matches(isDisplayed()))
+      onView(withId(R.id.onboarding_steps_count)).check(matches(not(isDisplayed())))
+      onView(withId(R.id.onboarding_navigation_back)).check(
+        matches(withEffectiveVisibility(Visibility.GONE))
+      )
+      onView(withId(R.id.onboarding_navigation_continue)).check(
+        matches(withEffectiveVisibility(Visibility.GONE))
+      )
+    }
+  }
+
+  @Test
+  fun testAudioLanguage_fromOptions_landscape_onboardingV2Enabled_navigationViewsAreNotDisplayed() {
+    launchV2FlowWithLanguage(ENGLISH_AUDIO_LANGUAGE, OPTIONS_SCREEN).use {
+      onView(withText("In Oppia, you can listen to lessons!")).check(matches(isDisplayed()))
+      onView(withText(context.getString(R.string.audio_language_fragment_subtitle)))
+        .check(matches(isDisplayed()))
+      onView(withId(R.id.audio_language_dropdown_list)).check(matches(isDisplayed()))
+      onView(withId(R.id.onboarding_steps_count)).check(matches(not(isDisplayed())))
+      onView(withId(R.id.onboarding_navigation_back)).check(
+        matches(withEffectiveVisibility(Visibility.GONE))
+      )
+      onView(withId(R.id.onboarding_navigation_continue)).check(
+        matches(withEffectiveVisibility(Visibility.GONE))
+      )
+    }
+  }
+
+  @Test
+  fun testAudioLanguage_fromOptions_onboardingV2Enabled_toolbarIsDisplayed() {
+    launchV2FlowWithLanguage(ENGLISH_AUDIO_LANGUAGE, OPTIONS_SCREEN).use {
+      onView(withId(R.id.reading_list_app_bar_layout)).check(matches(isDisplayed()))
+    }
+  }
+
+  @Test
+  fun testAudioLanguage_fromOnboarding_onboardingV2Enabled_toolbarIsNotDisplayed() {
+    launchV2FlowWithLanguage(ENGLISH_AUDIO_LANGUAGE, LEARNER_INTRO_SCREEN).use {
+      onView(withId(R.id.reading_list_app_bar_layout)).check(matches(not(isDisplayed())))
+    }
+  }
+
   @Test
   fun testFragment_portraitMode_backButtonPressed_currentScreenIsDestroyed() {
-    initializeTestApplicationComponent(enableOnboardingFlowV2 = true)
-    launch<AudioLanguageActivity>(
-      createDefaultAudioActivityIntent(ENGLISH_AUDIO_LANGUAGE)
-    ).use { scenario ->
+    launchV2FlowWithLanguage(ENGLISH_AUDIO_LANGUAGE, LEARNER_INTRO_SCREEN)
+      .use { scenario ->
+        onView(withId(R.id.onboarding_navigation_back)).perform(click())
+        testCoroutineDispatchers.runCurrent()
+        scenario.onActivity { activity ->
+          assertThat(activity.isFinishing).isTrue()
+        }
+      }
+  }
+
+  @Test
+  fun testFragment_landscapeMode_backButtonPressed_currentScreenIsDestroyed() {
+    launchV2FlowWithLanguage(ENGLISH_AUDIO_LANGUAGE, LEARNER_INTRO_SCREEN).use { scenario ->
+      rotateToLandscape()
```

Comment:
Nit here & elsewhere in this file: it's probably better to start the test in landscape mode for landscape (e.g. using test `@Config` qualifiers) tests since rotation doesn't actually change the orientation correctly currently (#2277 tracks this, though perhaps not well). The orientation change flow is still useful for verifying saved instance state cases, though.